sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/objects/objects.scala

```diff
@@ -1261,8 +1261,39 @@ case class InitializeJavaBean(beanInstance: Expression, setters: Map[String, Exp
   override def children: Seq[Expression] = beanInstance +: setters.values.toSeq
   override def dataType: DataType = beanInstance.dataType
 
-  override def eval(input: InternalRow): Any =
-    throw new UnsupportedOperationException("Only code-generated evaluation is supported.")
+  private lazy val resolvedSetters = {
+    val ObjectType(beanClass) = beanInstance.dataType
+    setters.map {
+      case (name, expr) =>
+        // Looking for known type mapping first, then using Class attached in `ObjectType`.
+        // Finally also looking for general `Object`-type parameter for generic methods.
+        val paramTypes = CallMethodViaReflection.typeMapping.getOrElse(expr.dataType,
```

Comment:
As #20753, we need to add other types into the mapping.

Reply:
Ok, who is making those changes? You or @kiszk?

Reply:
I may have no time to do tonight, if @kiszk do not make those changes, I will do it tomorrow.

Reply:
I added the similar variable `typeJavaMapping` and `typeJavaMapping`. This is because `typeMapping` is used for `CallMethodViaReflection` whose comment says `For now, only types defined in `Reflect.typeMapping` are supported (basically primitives and string) as input types`.

If we can expand this support to additional classes (e.g. `DateType`, `TimestampType`, `BinaryType`, and `CalendarIntervalType`), I can merge three variables into one `typeMapping`.

WDYT? @hvanhovell and @viirya

Reply:
Is there any special reason it only supports basically primitives and string?

Reply:
If we want to expand the support, maybe we can have another PR to expand it in `CallMethodViaReflection` and merge those variables.

Reply:
I see. Is it better to have separate map variables? cc @hvanhovell

Reply:
Sorry for not coming back to this sooner. AFAIK `CallMethodViaReflection` expression was only designed to work with a couple of primitives. I think we are looking for something a little bit more complete here, i.e. support all types in Spark SQL's type system. I also don't think that we should put the mappings in `CallMethodViaReflection` because the mapping is now using in more expressions, `ScalaReflection` is IMO a better place for this logic.

And finally which PR will implement this. cc @maropu for visibility.

Reply:
Done in #20753 and #20797